src/main/drivers/flash_m25p16.c

```diff
@@ -74,48 +71,64 @@ static flashGeometry_t geometry = {.pageSize = M25P16_PAGESIZE};
  */
 static bool couldBeBusy = false;
 
+
+static void m25p16_disable(busDevice_t *bus)
+{
+    IOHi(bus->busdev_u.spi.csnPin);
+    __NOP();
+}
+
+static void m25p16_enable(busDevice_t *bus)
+{
+    __NOP();
+    IOLo(bus->busdev_u.spi.csnPin);
+}
+
+static void m25p16_transfer(busDevice_t *bus, const uint8_t *txData, uint8_t *rxData, int len)
+{
+    m25p16_enable(bus);
+    spiTransfer(bus->busdev_u.spi.instance, txData, rxData, len);
+    m25p16_disable(bus);
+}
+
 /**
  * Send the given command byte to the device.
  */
-static void m25p16_performOneByteCommand(uint8_t command)
+static void m25p16_performOneByteCommand(busDevice_t *bus, uint8_t command)
```

Comment:
maybe
```
static void m25p16_performOneByteCommand(busDevice_t *bus, uint8_t command)
{
  m25p16_transfer(bus, &command, NULL, sizeof(command));
}

Reply:
That trades a minor code size saving for a minor performance loss. Not sure.

Reply:
Yes. But it removes one code path into lower layer - less places to introduce bugs in future
But it's minor change, it will work both ways